lib/validation.py

```diff
@@ -184,3 +185,37 @@ def is_valid_automation(automation):
     except ValidationError as e:
         return False, str(e)
     return True, None
+
+
+def parse_validation_error(data: Union[Dict, List], key: str, errors: List[Dict]) -> str:
```

Comment:
```suggestion
def parse_validation_error(data: Union[Dict, List], key: str, the_error: ValidationError) -> str:
    """
    Generates a human-readable HTML snippet detailing the validation error.
    
    :param data: The data that parsing raised an error.
    :param key: (what does this do? add docs)
    :param the_error: The raised error.
    """
    errors = the_error.errors()
```
Fill in the docstring around `key`, and I think this signature makes it clearer what the function expects - you should change `json.loads(e.json())` to just `e` in your invocations if you do this